docs/user/gcp/install_upi.md

```diff
@@ -659,6 +756,30 @@ grafana-openshift-monitoring.apps.your.cluster.domain.example.com
 prometheus-k8s-openshift-monitoring.apps.your.cluster.domain.example.com
 ```
 
+## Add the Ingress firewall rules (optional)
+If you removed the firewall configuration [earlier](#create-firewall-rules-and-iam-roles), you'll need to manually create some firewall rules for the ingress services.
+These rules would normally be created by the ingress controller via the gcp cloud provider.
+
+If you are installing into a [Shared VPC (XPN)][sharedvpc],
+use the `--account` and `--project` parameters to perform these actions in the host project.
+
+### Add the health check firewall rule.
+Add the firewall rule to allow the gce health checks to access all services.
+This enables the ingress load balancers to determine the health status of their instances.
+
+```sh
+gcloud compute firewall-rules create --allow='tcp:30000-32767,udp:30000-32767' --network="${CLUSTER_NETWORK}" --source-ranges='130.211.0.0/22,35.191.0.0/16,209.85.152.0/22,209.85.204.0/22' --target-tags="${INFRA_ID}-master,${INFRA_ID}-worker" ${INFRA_ID}-ingress-hc
+```
+
+### Add the service firewall rules.
+Add the ingress firewall rules for inbound service traffic.
+If you want your cluster to be private, you can use `--source-ranges=${NETWORK_CIDR}`.
+This rule may need to be updated accordingly when adding services on ports other than `tcp:80,tcp:443`.
+
+```sh
+gcloud compute firewall-rules create --allow='tcp:80,tcp:443' --network="${CLUSTER_NETWORK}" --source-ranges="0.0.0.0/0" --target-tags="${INFRA_ID}-master,${INFRA_ID}-worker" ${INFRA_ID}-ingress
+```
+
```

Comment:
afaik, the kube-controller-manger (gcp) emits events for the firewall rules that need to be created for **each** load balancer service.

so i think we should make it clear that the example here is opening up everything when technically they should be using those events to create specific ones. an example of what the event looks like and how to do that would be useful here.

Reply:
I'll make it clear this is opening everything up. I'm not sure what the event looks like because I did not see them in my testing.

Reply:
Reworked this section. PTAL 67658b5fca0aeaa92b80e428086fdd30037dfc9d